src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -1038,7 +1087,14 @@ internal static bool TryCreateGetOnlyCollectionDataContract(Type type, [NotNullW
             }
         }
 
-        internal static MethodInfo? GetTargetMethodWithName(string name, Type type, Type interfaceType)
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:MakeGenericMethod",
```

Comment:
Add a link to [https://github.com/mono/linker/issues/1731](https://github.com/mono/linker/issues/1731). We shouldn't need this suppression once that is fixed.

cc @vitek-karas

Reply:
It might be fixed already as I just removed the suppression and I didn't get any warnings now.

Reply:
Ok scratch that, seems like I didn't need it because I was using a preview4 SDK which probably have the linker fix, but the one we currently use doesn't seem to have it so I'll add the suppression back and a link to the issue.